From b37b03e815cd4f25af70a57c08449f8c719dcc6a Mon Sep 17 00:00:00 2001
From: Nicola Soranzo <nicola.soranzo@earlham.ac.uk>
Date: Mon, 10 Jul 2017 20:03:14 +0100
Subject: [PATCH] Python3 fix for plot-vcfstats

Fix for the error:
```
TypeError: 'range' object does not support item assignment
```
in generated `plot.py`.

Fix https://github.com/samtools/bcftools/issues/624 .
---
 misc/plot-vcfstats | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/misc/plot-vcfstats b/misc/plot-vcfstats
index 7172f05d..680e6ced 100755
--- a/misc/plot-vcfstats
+++ b/misc/plot-vcfstats
@@ -1590,8 +1590,8 @@ sub plot_substitutions
             \\tfig = plt.figure(figsize=($$opts{img_width},$$opts{img_height}))
             \\tcm  = mpl.cm.get_cmap('autumn')
             \\tn = 12
-            \\tcol = range(n)
-            \\tfor i in range(n): col[i] = cm(1.*i/n)
+            \\tcol = []
+            \\tfor i in range(n): col.append(cm(1.*i/n))
             \\tax1 = fig.add_subplot(111)
             \\tax1.bar([row[0] for row in dat], [row[2] for row in dat], color=col)
             \\tax1.set_ylabel('Count')
